frontend/src/metabase/reference/metrics/MetricDetailContainer.jsx

```diff
@@ -37,19 +32,39 @@ const mapDispatchToProps = {
 )
 export default class MetricDetailContainer extends Component {
   static propTypes = {
+    router: PropTypes.shape({
+      replace: PropTypes.func.isRequired,
+    }).isRequired,
     params: PropTypes.object.isRequired,
     location: PropTypes.object.isRequired,
     user: PropTypes.object.isRequired,
     metric: PropTypes.object.isRequired,
     metricId: PropTypes.number.isRequired,
     databaseId: PropTypes.number.isRequired,
-    isEditing: PropTypes.bool,
   };
 
+  constructor(props) {
+    super(props);
+    this.startEditing = this.startEditing.bind(this);
+    this.endEditing = this.endEditing.bind(this);
+  }
+
   async fetchContainerData() {
     await actions.wrappedFetchMetricDetail(this.props, this.props.metricId);
   }
 
+  startEditing() {
```

Comment:
If you change this to `startEditing = () => {` it's already bound to `this` and you don't need the constructor.

Reply:
Yeah, I wanted to keep following the same approach the rest of the code does. I found a lot of binds in class constructors, so I thought it's the preferred way to bind `this` here. Should we switch to an arrow function here?